apps/arweave/src/ar_node_worker.erl

```diff
@@ -1926,3 +1710,317 @@ dump_mempool(TXs, MempoolSize) ->
 			?LOG_ERROR([{event, failed_to_dump_mempool}, {reason, Reason}])
 	end.
 
+handle_found_solution(Args, PrevB, State) ->
+	{Source, Solution, PoACache, PoA2Cache} = Args,
+	#mining_solution{
+		last_step_checkpoints = LastStepCheckpoints,
+		mining_address = MiningAddress,
+		next_seed = NonceLimiterNextSeed,
+		next_vdf_difficulty = NonceLimiterNextVDFDifficulty,
+		nonce = Nonce,
+		nonce_limiter_output = NonceLimiterOutput,
+		partition_number = PartitionNumber,
+		poa1 = PoA1,
+		poa2 = PoA2,
+		preimage = SolutionPreimage,
+		recall_byte1 = RecallByte1,
+		recall_byte2 = RecallByte2,
+		solution_hash = SolutionH,
+		start_interval_number = IntervalNumber,
+		step_number = StepNumber,
+		steps = SuppliedSteps
+	} = Solution,
+	MerkleRebaseThreshold = ?MERKLE_REBASE_SUPPORT_THRESHOLD,
+
+	#block{ indep_hash = PrevH, timestamp = PrevTimestamp,
+			wallet_list = WalletList,
+			nonce_limiter_info = PrevNonceLimiterInfo } = PrevB,
```

Comment:
One change here is that we use `PrevB` instead of the block values cached in `node_state`, right? This is because teh values cached in `node_state` are based on the current Tip block - but if we're in the middle of rebasing onto a sibling, the new 'tip' may not be the cached tip, right?

Reply:
The previous block is always cached but this time it is not necessarily the tip. I keep the approach that the node never "shifts to the side" - every state change is always a cumulative difficulty increase; a rebase may be looked at as a regular reorg except for a mining solution is reused.